codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/SerializeMiddleware.java

```diff
@@ -0,0 +1,223 @@
+                        "output", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("SerializeOutput"),
+                        "metadata", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("Metadata"),
+                        "opName", this.serialName,
+                        "body", generateHandleSerializeBody())
+        );
+    }
+
+    private GoWriter.Writable generateHandleSerializeBody() {
+        return goTemplate("""
+                    $route:W
+    
+                    $serialize:W
+    
+                """,
+                MapUtils.of(
+                        "input", ctx.getSymbolProvider().toSymbol(input),
+                        "request", generator.getApplicationProtocol().getRequestType(),
+                        "route", handleHttp(),
+                        "serialize", handlePayload(),
+                        "errorf", GoStdlibTypes.Fmt.Errorf
+                ));
+    }
+
+    private GoWriter.Writable handleHttp() {
+        return goTemplate("""
+                    input, ok := in.Parameters.($input:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected input type %T", in.Parameters)
+                    }
+                    _ = input
+                
+                    req, ok := in.Request.($request:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected transport type %T", in.Request)
+                    }
+                
+                    req.Method = $methodPost:T
+                    req.URL.Path = "/service/$service:L/operation/$operation:L"
+                    req.Header.Set("smithy-protocol", $protocol:S)
+                
+                    $contentTypeHeader:W
+                    $acceptHeader:W
+                
+                """,
+        MapUtils.of(
+                "input", ctx.getSymbolProvider().toSymbol(input),
+                "request", generator.getApplicationProtocol().getRequestType(),
+                "methodPost", GoStdlibTypes.Net.Http.MethodPost,
+                "service", ctx.getService().getId().getName(),
+                "operation", operation.getId().getName(),
+                "protocol", generator.getProtocol(),
+                "contentTypeHeader", setContentTypeHeader(),
+                "acceptHeader", acceptHeader(),
+                "serialize", handlePayload(),
+                "errorf", GoStdlibTypes.Fmt.Errorf
+                ));
+    }
+
+    private GoWriter.Writable handlePayload() {
+        return goTemplate("""
+            
+                jsonEncoder := $encoder:T()
+            
+                err = $serialize:L(input, jsonEncoder.Value)
+                if err != nil {
+                    return out, metadata, &$error:T{Err: err}
+                }
+
+            
+                payload := $reader:T(jsonEncoder.Bytes())
+                if req, err = req.SetStream(payload); err != nil {
```

Comment:
This is a little confusing to read because of the compound assign-if clause - you have to remember that `req` was created earlier above in the serialize middleware (and in another function in the generator)

Prefer something like

```go
req, err = req.SetStream(payload)
if err != nil {
  // ...
}
```

We generally use assign-ifs to create temporaries that only exist in the scope of that if statement.